server/pulp/server/db/model/__init__.py

```diff
@@ -49,6 +52,25 @@ def __init__(self, *args, **kwargs):
 
     meta = {'abstract': True}
 
+    def clean(self):
```

Comment:
Is this function called automatically by mongoengine at document creation time? I am not sure that I like the idea raising a `ValidationError` if it is the document that is broken rather than an instance.